File: Week-1/Solutions/AryaCodesAI/2-power-gauntlet.py
# Check if n crystals can power up the gauntlet
# Power crystals must be exactly 1, 2, 4, 8, 16... (powers of 2)

n = int(input())

# A number is a power of 2 if it has exactly one bit set in binary
# Using bitwise operation: n & (n-1) == 0 for powers of 2
if n > 0 and (n & (n - 1)) == 0:
    print("Yes")
else:
    print("No")

File: Week-1/Solutions/shrinivasshah/2-power-gauntlet.ts

function canPowerup (n: number): boolean {
    while (n > 1) {
        if (n % 2 !== 0) return false
        n = n / 2;
    }
    return true;
}

(console.log(canPowerup(8))); // true
(console.log(canPowerup(7))); // false
